engine/src/main/java/org/hibernate/validator/internal/properties/javabean/JavaBeanExecutable.java

```diff
@@ -212,21 +194,43 @@ public String toString() {
 		);
 	}
 
-	private static Type getParameterGenericType(Type[] parameterTypes, Type[] genericParameterTypes, int parameterIndex) {
-		// getGenericParameterTypes() doesn't return synthetic parameters; in this case fall back to getParameterTypes()
-		Type[] typesToConsider;
-		if ( parameterIndex >= genericParameterTypes.length ) {
-			typesToConsider = parameterTypes;
-		}
-		else {
-			typesToConsider = genericParameterTypes;
+	private static List<JavaBeanParameter> getParameters(Executable executable) {
+		if ( executable.getParameterCount() == 0 ) {
+			return Collections.emptyList();
 		}
 
-		Type type = typesToConsider[parameterIndex];
+		List<JavaBeanParameter> parameters = new ArrayList<>( executable.getParameterCount() );
 
-		if ( type instanceof TypeVariable ) {
-			type = TypeHelper.getErasedType( type );
```

Comment:
So the reason is I don't think the `type` of a parameter could ever be a `TypeVariable` as `TypeVariable` is supposed to be used for type arguments.

That's why I removed this part.

Reply:
I think it can, in this case:

```
public class Foo {
   public <T> bar(T param);
}
```

Or in this one:

```
public class Foo<T> {
   public bar(T param);
}
```

The type of `param` is simply `T`, which is a `TypeVariable`.

Reply:
Of course this can also happen with wildcard, where the erasure would make more sense:

```
public class Foo<T extends Number> {
   public void bar(T param) { }
}
```

Reply:
Hum, good call, will fix that.

Reply:
FWIW, I tried to add a test for that but the type is later erased again when used.

I think I will still keep the erasure here though as it makes sense.